### PR TITLE
feat: Support the Backend resource for Ext-Auth

### PR DESCRIPTION
**Pull request by zirain**
xref: https://github.com/envoyproxy/gateway/issues/3665

### CODE REVIEW

File: internal/gatewayapi/securitypolicy.go [hunk 1]
```diff
@@ -892,6 +889,31 @@ func (t *Translator) buildExtAuth(policy *egv1a1.SecurityPolicy, resources *Reso
...
+	if backendRef == nil {
+		return ""
+	}
+
+	backendNamespace := NamespaceDerefOr(backendRef.Namespace, policy.Namespace)
+	backendKind := KindDerefOr(backendRef.Kind, KindService)
+	if backendKind == egv1a1.KindBackend {
+		backend := resources.GetBackend(backendNamespace, string(backendRef.Name))
+		if backend != nil {
+			// TODO: exists multi FQDN endpoints?
```
Comment (guydc) [inline internal/gatewayapi/securitypolicy.go:902]: I think that we need to support user-defined authority for cases where there are multiple endpoints. We can leave this to a separate PR.

**Review by guydc — APPROVED** [review]
LGTM, Thanks!

**Review by arkodg — APPROVED** [review]
thanks !

File: internal/gatewayapi/securitypolicy.go (hunk 1 above)
Comment (zirain) [reply internal/gatewayapi/securitypolicy.go:902]: yes, telemetry also need it.
